dpcpp/factorization/factorization_kernels.dp.cpp

```diff
@@ -0,0 +1,146 @@
+
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+
+
+#include "core/components/prefix_sum.hpp"
+#include "core/matrix/csr_builder.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace dpcpp {
+/**
+ * @brief The factorization namespace.
+ *
+ * @ingroup factor
+ */
+namespace factorization {
+
+
+template <bool IsSorted, typename ValueType, typename IndexType>
+void find_missing_diagonal_elements(
+    const matrix::Csr<ValueType, IndexType> *mtx,
+    IndexType *elements_to_add_per_row, bool *changes_required)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+
+template <typename ValueType, typename IndexType>
+void add_missing_diagonal_elements(const matrix::Csr<ValueType, IndexType> *mtx,
+                                   ValueType *new_values,
+                                   IndexType *new_col_idxs,
+                                   const IndexType *row_ptrs_addition)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+
+}  // namespace factorization
+
+
+template <typename ValueType, typename IndexType>
+void add_diagonal_elements(std::shared_ptr<const DpcppExecutor> exec,
+                           matrix::Csr<ValueType, IndexType> *mtx,
+                           bool is_sorted)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_FACTORIZATION_ADD_DIAGONAL_ELEMENTS_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void initialize_row_ptrs_l_u(
+    std::shared_ptr<const DpcppExecutor> exec,
+    const matrix::Csr<ValueType, IndexType> *system_matrix,
+    IndexType *l_row_ptrs, IndexType *u_row_ptrs)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_FACTORIZATION_INITIALIZE_ROW_PTRS_L_U_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void initialize_l_u(std::shared_ptr<const DpcppExecutor> exec,
+                    const matrix::Csr<ValueType, IndexType> *system_matrix,
+                    matrix::Csr<ValueType, IndexType> *csr_l,
+                    matrix::Csr<ValueType, IndexType> *csr_u)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_FACTORIZATION_INITIALIZE_L_U_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void initialize_row_ptrs_l(
+    std::shared_ptr<const DpcppExecutor> exec,
+    const matrix::Csr<ValueType, IndexType> *system_matrix,
+    IndexType *l_row_ptrs)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_FACTORIZATION_INITIALIZE_ROW_PTRS_L_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void initialize_l(std::shared_ptr<const DpcppExecutor> exec,
+                  const matrix::Csr<ValueType, IndexType> *system_matrix,
+                  matrix::Csr<ValueType, IndexType> *csr_l, bool diag_sqrt)
+{
+    GKO_NOT_IMPLEMENTED;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_FACTORIZATION_INITIALIZE_L_KERNEL);
+
+
```

Comment:
```suggestion

}  // namespace factorization
```
the factorization namespace should be here